pkg/ottl/parser.go

```diff
@@ -306,3 +306,50 @@ func (s *Statements[K]) Eval(ctx context.Context, tCtx K) (bool, error) {
 	}
 	return false, nil
 }
+
+// ConditionSequence represents a list of Condition that will be executed sequentially for a TransformContext
+// and will handle errors returned by conditions based on an ErrorMode.
+type ConditionSequence[K any] struct {
+	conditions        []*Condition[K]
+	errorMode         ErrorMode
+	telemetrySettings component.TelemetrySettings
+}
+
+type ConditionSequenceOption[K any] func(*ConditionSequence[K])
+
+// NewConditionSequence creates a new ConditionSequence with the provided Condition slice, ErrorMode, and component.TelemetrySettings.
+// You may also augment the ConditionSequence with a slice of ConditionSequenceOption.
+func NewConditionSequence[K any](conditions []*Condition[K], errorMode ErrorMode, telemetrySettings component.TelemetrySettings, options ...ConditionSequenceOption[K]) ConditionSequence[K] {
```

Comment:
Two questions on the function signature:
1. Can `errorMode` be an option instead of required? I think in the future we should select a default so users have a default expectation of how errors are handled in OTTL statements in any component.
2. If we want to make it required, is the intention also to require it in `NewStatements` (or `NewStatementSequence`)? It would be nice if the two APIs matched.

Reply:
I've been noodling on this for a while.  On one had something like `errorMode` is very natural as an Option, which is why we did it for `NewStatements`.  But when I reflected on why we created `Statements`, its sole purpose was to help us with handling the error responses while processing a sequence of statements. Since we created it to handle different error modes, I like the idea of the errorMode for a `Sequence` being explicitly set by the user.

I can see an argument for still allowing it to be an option so it is easy to set a default value. If we go this route we'll have to make sure the Option for a `ConditionSequence` doesn't conflict with the Option for a `StatementSequence`, either by unique names or maybe some fancy generics solution.

In the end I want `Statements` to be renamed to `StatementSequence` and to match the API we decide on for `ConditionSequence`.  I'll do that in a followup PR.

Reply:
That all makes sense.  I don't feel too strongly about this, so I've switched it back to being an Option, and set `Ignore` as the default value (but I'm ok switching it to `Propagate`).

Reply:
Setting `Ignore` as the default value would make this a breaking change in the components that use OTTL for selection, right? I'm thinking we should probably make it `Propagate` for now, then announce it as a breaking change for all affected components when we make that switch.

Reply:
My bad, I see now. `internal/filter` requires an error mode, and the components set it to `Propagate` themselves, so the end behavior would not change. I'm good either way in that case.

Reply:
I think you're right that `Propagate` is a better default value.  It ensures that users never miss an error unless they intend to.